src/EditorFeatures/Core/StackTraceExplorer/StackTraceExplorerUtilities.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.StackTraceExplorer
+{
+    internal static class StackTraceExplorerUtilities
+    {
+        public static async Task<DefinitionItem?> GetDefinitionAsync(Solution solution, StackFrameCompilationUnit compilationUnit, StackFrameSymbolPart symbolPart, CancellationToken cancellationToken)
+        {
+            // MemberAccessExpression is [Expression].[Identifier], and Identifier is the 
+            // method name.
+            var typeExpression = compilationUnit.MethodDeclaration.MemberAccessExpression.Left;
+            var fullyQualifiedTypeName = typeExpression.ToString();
+            var typeName = typeExpression is StackFrameQualifiedNameNode qualifiedName
+                ? qualifiedName.Right.ToString()
+                : typeExpression.ToString();
+
+            RoslynDebug.AssertNotNull(fullyQualifiedTypeName);
+
+            var methodIdentifier = compilationUnit.MethodDeclaration.MemberAccessExpression.Right;
+            var methodTypeArguments = compilationUnit.MethodDeclaration.TypeArguments;
+            var methodArguments = compilationUnit.MethodDeclaration.ArgumentList;
+
+            var methodName = methodIdentifier.ToString();
+
+            //
+            // Do a first pass to find projects with the type name to check first 
+            //
+            using var _ = PooledObjects.ArrayBuilder<Project>.GetInstance(out var candidateProjects);
+            foreach (var project in solution.Projects)
+            {
+                if (!project.SupportsCompilation)
+                {
+                    continue;
+                }
+
+                var containsSymbol = await project.ContainsSymbolsWithNameAsync(
+                    (name) => name == typeName,
```

Comment:
```suggestion
                    typeName,
```
?

Reply:
Technically that method doesn't exist, but I don't mind adding it.